src/components/Reown/NanoContract/BaseNanoContractRequest.js

```diff
@@ -0,0 +1,526 @@
+    dispatch(statusConfig.setReadyAction());
+
+    // Do nothing if nano contract is not registered and don't call initialize method
+    if (notRegistered) return undefined;
+
+    if (!blueprintInfo) {
+      dispatch(nanoContractBlueprintInfoRequest(nano.blueprintId));
+    }
+
+    // Request token data for each unknown token present in actions
+    const unknownTokensUid = [];
+    const actionTokensUid = nano.actions?.map((each) => each.token) || [];
+    actionTokensUid.forEach((uid) => {
+      if (uid !== DEFAULT_TOKEN.uid && !(uid in knownTokens)) {
+        unknownTokensUid.push(uid);
+      }
+    });
+
+    if (unknownTokensUid.length > 0) {
+      dispatch(unregisteredTokensDownloadRequest({ uids: unknownTokensUid }));
+    }
+
+    // Clean up function
+    return () => {
+      // Restore ready status to Nano Contract registration state if
+      // we have had a registration while handling a new transaction request
+      dispatch(nanoContractRegisterReady());
+      dispatch(statusConfig.setReadyAction());
+      // Dismiss the retry condition to New Nano Contract Transaction
+      dispatch(newNanoContractRetryDismiss());
+      // If the user leaves without accepting or declining, we should decline the transaction
+      if (status !== statusConfig.statusConstants.SUCCESSFUL) {
+        dispatch(reownReject());
+      }
+    };
+  }, [notRegistered]);
+
+  const toggleSelectAddressModal = () => {
+    setShowSelectAddressModal(!showSelectAddressModal);
+  };
+
+  const handleAddressSelection = (newAddress) => {
+    setNcAddress(newAddress);
+    toggleSelectAddressModal();
+  };
+
+  const onAcceptTransaction = () => {
+    if (!ncAddress) {
+      console.warn('No address selected, cannot submit transaction');
+      return;
+    }
+
+    dispatch(statusConfig.setLoadingAction());
+
+    const acceptedData = prepareAcceptData
+      ? prepareAcceptData(nanoWithCaller, ncAddress)
+      : nanoWithCaller;
+
+    if (typeof onAccept === 'function') {
+      onAccept(acceptedData);
+    } else {
+      dispatch(reownAccept(acceptedData));
+    }
+  };
+
+  const onRegisterNanoContract = () => {
+    dispatch(nanoContractRegisterRequest({
+      address: firstAddress.address,
+      ncId: nanoWithCaller.ncId
+    }));
+  };
+
+  const onDeclineTransaction = () => {
+    setShowDeclineModal(true);
+  };
+
+  const { navigateBack } = useBackButtonHandler(
+    onDeclineTransaction,
+    status === statusConfig.statusConstants.SUCCESSFUL
+  );
+
+  const onDeclineConfirmation = () => {
+    setShowDeclineModal(false);
+    dispatch(reownReject());
+
+    if (typeof onReject === 'function') {
+      onReject();
+    }
+
+    navigateBack();
+  };
+
+  const onDismissDeclineModal = () => {
+    setShowDeclineModal(false);
+  };
+
+  const onFeedbackModalDismiss = () => {
+    dispatch(statusConfig.retryDismissAction());
+    navigateBack();
+  };
+
+  const onTryAgain = () => {
+    dispatch(statusConfig.retryAction());
```

Comment:
We also had `dispatch(setNewNanoContractStatusReady());` in the other component. Don't we need it anymore?

Reply:
We do, re-added it 

thanks